spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -64,13 +75,18 @@
     this.writeInfo = info;
     this.dsSchema = info.schema();
     this.overwriteMode = writeConf.overwriteMode();
+    this.rewrittenFileSetId = writeConf.rewrittenFileSetId();
     this.handleTimestampWithoutZone = writeConf.handleTimestampWithoutZone();
+    this.requestDistributionAndOrdering = allIdentityTransforms(table.spec());
```

Comment:
I'll add a check if extensions are enabled later. The plan is to have a rule that would build Iceberg writes and translate all connector transforms to Iceberg transforms.

Reply:
Should add some warning logs here?

Reply:
Yeah, that's a good idea. I'll add them to the methods that build the distribution and ordering.

Reply:
Added a warning.